src/gui/optionsdialog.cpp

```diff
@@ -1445,34 +1444,32 @@ void OptionsDialog::toggleComboRatioLimitAct()
 
 void OptionsDialog::enableProxy(int index)
 {
-    if (index) {
+    if (index != 0) { // Any proxy type is used
         //enable
         m_ui->lblProxyIP->setEnabled(true);
         m_ui->textProxyIP->setEnabled(true);
         m_ui->lblProxyPort->setEnabled(true);
         m_ui->spinProxyPort->setEnabled(true);
         m_ui->checkProxyPeerConnecs->setEnabled(true);
-        if (index > 1) {
+        if (index > 1) { // SOCKS5 or HTTP
             m_ui->checkProxyAuth->setEnabled(true);
             m_ui->isProxyOnlyForTorrents->setEnabled(true);
         }
         else {
             m_ui->checkProxyAuth->setEnabled(false);
             m_ui->checkProxyAuth->setChecked(false);
-            m_ui->isProxyOnlyForTorrents->setEnabled(false);
             m_ui->isProxyOnlyForTorrents->setChecked(true);
```

Comment:
While we're here, is it expected to check this only when selecting SOCKS4? I guess not because it is disabled for SOCKS4 and also it doesn't happen in WebUI.

Reply:
Never mind i found this which i suppose means that WebUI should be adjusted
```
bool ProxyConfigurationManager::isProxyOnlyForTorrents() const
{
    return m_isProxyOnlyForTorrents || (m_config.type == ProxyType::SOCKS4);
}
```

Reply:
I'm sorry, I'm off topic... Is SOCKS4 still relevant? Maybe we should stop supporting it? This will make the logic less confusing.

Reply:
If it's not broken and libt still supports it why remove it? Anyway don't ask me about this.